### PR TITLE
feat(uri-handler): support other bitcoin-like coins in desktop app

### LINKED ISSUE #5241: LTC address scan from QR not working
**Describe the bug**
LTC address scan is not working when sending to mobile Coinomi wallet.

**Info:**

-   Remembered Wallet: no
-   Suite Version: app version 22.3.2.16437
-   OS: MacOS, 12.3.1 (21E258)

**How to reproduce**

1. Go to Wallet
2. Select Litecoin token
3. Click on Send
4. Scan address from QR code
6. Receiver address won't show in the input box

**Expected behavior**
Address should be present in the input.

### LINKED ISSUE #5266: Extend URI handler for altcoins
Extend solution of https://github.com/trezor/trezor-suite/issues/3294
When user scans address with some of the listed prefix → system should open Trezor Suite and fill in the `send to` address

Litecoin `litecoin:`
Ethereum `ethereum:`
Ethereum Classic `ethclassic:`
XRP `ripple:`
Bitcoin Cash `bitcoincash:`
Bitcoin Gold `bitcoingold:`
Dash `dash:`
DigiByte `digibyte:`
Dogecoin `dogecoin:`
Namecoin `namecoin:`
Vertcoin `vertcoin:`
Zcash `zcash:`
Cardano `cardano:`

Related to: https://github.com/trezor/trezor-suite/issues/5241

### PR DESCRIPTION
**Pull request by tomasklim**
close #5266
close #5241

Bitcoin-like coins' uri handler will now work in desktop app. 
```
    "litecoin",
    "bitcoincash",
    "bitcoingold",
    "dash",
    "digibyte",
    "dogecoin",
    "namecoin",
    "vertcoin",
    "zcash"
```

Non-bitcoin-like coins could have different implementation. Therefore they are not part of this PR and can be done later.
e.g. https://eips.ethereum.org/EIPS/eip-681

This PR also fixes scanning QR code. `scheme:address?amount=x` 
For non-bitcoin-like coins, valid qr codes will fill only address in case `amount` is not part of their specification. 

Follow-up issue:
#5508

### CODE REVIEW

File: packages/suite/src/constants/suite/protocol.ts [hunk 1]
```diff
@@ -2,8 +2,35 @@ import type { Network } from '@wallet-types';
...
+    DIGIBYTE = 'digibyte',
+    DOGECOIN = 'dogecoin',
+    NAMECOIN = 'namecoin',
+    VERTCOIN = 'vertcoin',
+    ZCASH = 'zcash',
+    // only bitcoin-like coins are supported in uri handler, others are required for QR code scan
+    ETHEREUM = 'ethereum',
+    ETHEREUM_CLASSIC = 'ethclassic',
+    XRP = 'ripple',
+    CARDANO = 'cardano:',
```
Comment (marekrjpolak) [inline packages/suite/src/constants/suite/protocol.ts:18]: Should the `:` be there?

Comment (marekrjpolak) [inline packages/suite/src/constants/suite/protocol.ts]: Should the `:` be there?

**Review by marekrjpolak — COMMENTED** [review]
Looks good and QR scanning works, but I'm unfortunately not able to properly test the scheme registration in desktop Suite on NixOS.

By the way, when scanning QR code including URI scheme, it is now possible to verify also that the address was scanned in the right account's send form. But as it's still possible to scan it without the scheme and not be able to do this check, it doesn't necessarily have to be part of this PR imho.

File: packages/suite/src/constants/suite/protocol.ts (hunk 1 above)
Comment (tomasklim) [reply packages/suite/src/constants/suite/protocol.ts]: 4d0bbefb3a4ba7aa79856796c897b31166322177

File: packages/suite/src/utils/suite/__tests__/protocol.test.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { getProtocolInfo, isProtocolScheme } from '@suite-utils/protocol';
+import * as fixtures from '../__fixtures__/protocol';
+
+describe('getProtocolInfo', () => {
+    fixtures.getProtocolInfo.forEach(f => {
```
Comment (dahaca) [inline packages/suite/src/utils/suite/__tests__/protocol.test.ts:5]: nit: would it makes sense to not use one letter variables even here? 😅 Or better stop using them altogether. OR outline where they can be use and where they shouldn't be.

File: packages/suite/src/utils/suite/protocol.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    address: string;
+    amount?: number;
+};
+
+export const isProtocolScheme = (scheme: string): scheme is PROTOCOL_SCHEME =>
+    Object.values(PROTOCOL_SCHEME).includes(scheme as PROTOCOL_SCHEME);
+
+export const getProtocolInfo = (uri: string): CoinProtocolInfo | null => {
+    const url = parseUri(uri);
+    if (!url) return null;
```
Comment (dahaca) [inline packages/suite/src/utils/suite/protocol.ts]: nit: I think we agreed on not using one line ifs on previous code improvements 👀

File: packages/suite/src/utils/suite/protocol.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    const { protocol, pathname, host, search } = url;
+    const scheme = protocol.slice(0, -1); // slice ":" from protocol
```
Comment (dahaca) [inline packages/suite/src/utils/suite/protocol.ts]: Perhaps better use `replace(':','')`?

File: packages/suite/src/utils/suite/__tests__/protocol.test.ts (hunk 2 above)
Comment (tomasklim) [reply packages/suite/src/utils/suite/__tests__/protocol.test.ts:5]: I moved it like that from another file 🤯
Screenshot 2022-05-30 at 11 14 22

File: packages/suite/src/utils/suite/protocol.ts (hunk 4 above)
Comment (tomasklim) [reply packages/suite/src/utils/suite/protocol.ts]: 🤷‍♂️  We know that protocol is always `xxx:` and we need to get rid of the last char.

File: packages/suite/src/utils/suite/protocol.ts (hunk 3 above)
Comment (tomasklim) [reply packages/suite/src/utils/suite/protocol.ts]: 1e7b694621bbadbc30dbac7245b55b430a214043

File: packages/suite/src/utils/suite/__tests__/protocol.test.ts (hunk 2 above)
Comment (tomasklim) [reply packages/suite/src/utils/suite/__tests__/protocol.test.ts:5]: We have 253 occurrences. I do not think that it is a good idea to change it.

**Comment by tomasklim** [comment]
> By the way, when scanning QR code including URI scheme, it is now possible to verify also that the address was scanned in the right account's send form. But as it's still possible to scan it without the scheme and not be able to do this check, it doesn't necessarily have to be part of this PR IMHO.

724f71ed0c645d4f52d9a1d5f2b91356339e46d3

File: packages/suite/src/utils/suite/__tests__/protocol.test.ts (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/utils/suite/__tests__/protocol.test.ts:5]: Of course! I didn't know it's a "pattern", let it be.

**Review by dahaca — APPROVED** [review]

**Pull request merged**